tests/unit/test_run.py

```diff
@@ -0,0 +1,1788 @@
+        test_init = test_init.drop("year", axis="columns")
+    elif in_format == "time_col_index":
+        idx = ["climate_model", "model", "scenario", "region", "variable", "unit"]
+        test_init = test_pd_run_df.melt(id_vars=idx, var_name="year")
+        test_init["time"] = test_init["year"].apply(lambda x: dt.datetime(x, 1, 1))
+        test_init = test_init.drop("year", axis="columns")
+        test_init = test_init.set_index(idx + ["time"])
+    elif in_format == "time_col_str_simple":
+        idx = ["climate_model", "model", "scenario", "region", "variable", "unit"]
+        test_init = test_pd_run_df.melt(id_vars=idx, var_name="year")
+        test_init["time"] = test_init["year"].apply(
+            lambda x: "{}-1-1 00:00:00".format(x)
+        )
+        test_init = test_init.drop("year", axis="columns")
+    elif in_format == "time_col_str_complex":
+        idx = ["climate_model", "model", "scenario", "region", "variable", "unit"]
+        test_init = test_pd_run_df.melt(id_vars=idx, var_name="year")
+        test_init["time"] = test_init["year"].apply(lambda x: "{}/1/1".format(x))
+        test_init = test_init.drop("year", axis="columns")
+    elif in_format == "str_times":
+        test_init = test_pd_run_df.copy()
+        test_init.columns = test_init.columns.map(
+            lambda x: "{}/1/1".format(x) if isinstance(x, int) else x
+        )
+
+    res = ScmRun(test_init)
+    assert (res["year"].unique() == [2005, 2010, 2015]).all()
+    assert (
+        res["time"].unique()
+        == [dt.datetime(2005, 1, 1), dt.datetime(2010, 1, 1), dt.datetime(2015, 1, 1)]
+    ).all()
+
+    res_df = res.timeseries()
+    res_df.columns = res_df.columns.map(lambda x: x.year)
+    res_df = res_df.reset_index()
+
+    pd.testing.assert_frame_equal(
+        res_df[test_pd_run_df.columns.tolist()], test_pd_run_df, check_like=True
+    )
+
+
+def test_init_df_missing_time_axis_error(test_pd_df):
+    idx = ["climate_model", "model", "scenario", "region", "variable", "unit"]
+    test_init = test_pd_df.melt(id_vars=idx, var_name="year")
+    test_init = test_init.drop("year", axis="columns")
+    error_msg = re.escape("invalid time format, must have either `year` or `time`!")
+    with pytest.raises(ValueError, match=error_msg):
+        ScmDataFrame(test_init)
+
+
+def test_init_df_missing_time_columns_error(test_pd_df):
+    test_init = test_pd_df.copy()
+    test_init = test_init.drop(
+        test_init.columns[test_init.columns.map(lambda x: isinstance(x, int))],
+        axis="columns",
+    )
+    error_msg = re.escape(
+        "invalid column format, must contain some time (int, float or datetime) columns!"
+    )
+    with pytest.raises(ValueError, match=error_msg):
+        ScmDataFrame(test_init)
+
+
+def test_init_df_missing_col_error(test_pd_df):
+    test_pd_df = test_pd_df.drop("model", axis="columns")
+    error_msg = re.escape("missing required columns `['model']`!")
```

Comment:
Is it worth also enforcing `climate_model` being a column? Just to make sure that there is always both `model` and `climate_model` and people understand the difference between the two?

Reply:
As discussed, don't enforce here but allow people to subclass `ScmRun` with whatever enforcements they require

Reply:
@lewisjared do we need a new issue so we can address this at a later point?

Reply:
#42